### PR TITLE
Port over VSH1's content precaching/addtodownloads stocks + update Vagineer/Hale models

### LINKED ISSUE #26: Use new Hale/Vagineer models by default
Why not just use the newer models by default? It seems kind of strange to be supporting the older models at all, considering you can't even headshot the old Vagineer model.

### LINKED ISSUE #27: Switch to VSH1's method of precaching custom content
I just realized VSH2 isn't using VSH1's PrepareModel/Material/Sound (and other such functions) stocks, which combine adding to the downloads table and precaching into one function.
While I'm addressing #26 I'll work this as well.

### PR DESCRIPTION
**Pull request by Starblaster64**
Presently pending review before merging.
Resolves #26 
Resolves #27

### CODE REVIEW

**Comment by Starblaster64** [comment]
Ignore the Travis build for now, TF2Items/SteamTools repo is still down.

File: scripting/modules/bosses/bunny.sp [hunk 1]
```diff
@@ -345,73 +365,27 @@ public CBunny ToCBunny (const BaseBoss guy)
 
 public void AddBunnyToDownloads()
 {
-	char s[PLATFORM_MAX_PATH];
+	// static const char s[PLATFORM_MAX_PATH]; //This is the odd one out of all the boss.sp files, is it the rule or exception?
```
Comment (Starblaster64) [inline scripting/modules/bosses/bunny.sp]: Ignore this line comment, I derped and didn't realise I was the one who find+replaced in static const.

**Comment by assyrianic** [comment]
did this compile fully and I'm assuming the test downloading worked?

**Comment by Starblaster64** [comment]
Yes, I did test it locally.

**Pull request merged**